components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CertificateMgtUtils.java

```diff
@@ -141,6 +160,64 @@ public ResponseCode addCertificateToTrustStore(String base64Cert, String alias)
         return responseCode;
     }
 
+    /**
+     * To validate the current certificate and alias.
+     *
+     * @param alias       Alias of the certificate.
+     * @param certificate Bas64 endcoded certificated.
+     * @return response code based on the validation
+     */
+    public ResponseCode validateCertificate(String alias, int tenantId, String certificate) {
+        File trustStoreFile = new File(TRUST_STORE);
+        ResponseCode responseCode = ResponseCode.SUCCESS;
+        ByteArrayInputStream serverCert = null;
+
+        try {
+            localTrustStoreStream = new FileInputStream(trustStoreFile);
+            KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            trustStore.load(localTrustStoreStream, TRUST_STORE_PASSWORD);
+            if (StringUtils.isNotEmpty(alias) && trustStore.containsAlias(alias + "_" + tenantId)) {
+                responseCode = ResponseCode.ALIAS_EXISTS_IN_TRUST_STORE;
+            }
+            if (responseCode != ResponseCode.ALIAS_EXISTS_IN_TRUST_STORE) {
+                byte[] cert = (Base64.decodeBase64(certificate.getBytes(StandardCharsets.UTF_8)));
+                serverCert = new ByteArrayInputStream(cert);
+
+                if (serverCert.available() == 0) {
+                    responseCode = ResponseCode.CERTIFICATE_NOT_FOUND;
+                } else {
+                    CertificateFactory cf = CertificateFactory.getInstance(CERTIFICATE_TYPE);
+                    while (serverCert.available() > 0) {
+                        Certificate generatedCertificate = cf.generateCertificate(serverCert);
+                        X509Certificate x509Certificate = (X509Certificate) generatedCertificate;
+                        if (x509Certificate.getNotAfter().getTime() <= System.currentTimeMillis()) {
+                            responseCode = ResponseCode.CERTIFICATE_EXPIRED;
+                        }
+                    }
+                }
+            }
+        } catch (IOException e) {
+            log.error("I/O Exception while trying to load trust store while trying to check whether alias " + alias
+                    + " exists", e);
+            responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (CertificateException e) {
+            log.error("Certificate Exception while trying to load trust store while trying to check whether alias "
+                    + alias + " exists", e);
+            responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (NoSuchAlgorithmException e) {
+            log.error("No Such Algorithm Exception while trying to load trust store while trying to check whether "
```

Comment:
Thanks for the suggestion on this. We have catched the NoSuchAlgorithmException at this point and in the error message we are try to tell the user on at which scenario this exception has occured. Regarding why has it occured and how to overcome this error, I think for that User can refer the oracle documentation on "NoSuchAlgorithmException" at [1] https://docs.oracle.com/javase/8/docs/api/java/security/NoSuchAlgorithmException.html. If we are trying to do include the same information here, it will be more of a duplicate. WDYT?